core/src/main/java/com/linecorp/armeria/client/endpoint/dns/DnsAddressEndpointGroup.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.endpoint.dns;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+import java.net.InetAddress;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nullable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.client.endpoint.DynamicEndpointGroup;
+import com.linecorp.armeria.common.CommonPools;
+
+import io.netty.channel.EventLoop;
+import io.netty.handler.codec.dns.DnsRecordType;
+import io.netty.resolver.NameResolver;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.ScheduledFuture;
+
+/**
+ * {@link DynamicEndpointGroup} which resolves targets using DNS address queries (A and AAAA). This is useful
+ * for environments where service discovery is handled using DNS - for example, Kubernetes uses SkyDNS for
+ * service discovery.
+ */
+public class DnsAddressEndpointGroup extends DynamicEndpointGroup implements AutoCloseable {
+
+    /**
+     * Creates a {@link DnsAddressEndpointGroup} with an unspecified port that schedules queries on a random
+     * {@link EventLoop} from {@link CommonPools#workerGroup()} every 1 second.
+     *
+     * @param hostname the hostname to query DNS queries for.
+     */
+    public static DnsAddressEndpointGroup of(String hostname) {
+        return DnsAddressEndpointGroup.of(hostname, 0);
+    }
+
+    /**
+     * Creates a {@link DnsAddressEndpointGroup} that schedules queries on a random {@link EventLoop} from
+     * {@link CommonPools#workerGroup()} every 1 second.
+     *
+     * @param hostname the hostname to query DNS queries for.
+     * @param defaultPort the port to use when the DNS answer does not contain one. {@code 0} indicates an
```

Comment:
nit: the explanation about `defaultPort` is a little bit different from other two java-docs. If you intend this, never mind.